internal/tidy/tidy_test.go

```diff
@@ -0,0 +1,122 @@
+package tidy_test
+
+import (
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/engine/internal/tidy"
+	"sort"
+	"testing"
+)
+
+func TestUnnestLongerSortedHappy(t *testing.T) {
+	astromech := []string{"q7", "bb", "r2", "r4"}
+	protocol := []string{"c3po", "000", "talky"}
+	battle := []string{"b1", "ig"}
+	probe := []string{"viper"}
+	astromechGroup := "astromech"
+	protocolGroup := "protocol"
+	battleGroup := "battle"
+	probeGroup := "probe"
+	astromechSorted := make([]string, len(astromech))
+	protocolSorted := make([]string, len(protocol))
+	battleSorted := make([]string, len(battle))
+	probeSorted := make([]string, len(probe))
+	sort.Strings(astromechSorted)
+	sort.Strings(battleSorted)
+	sort.Strings(probeSorted)
+	sort.Strings(protocolSorted)
+	expOut := [][]string{
+		{"astromech", "bb"},
+		{"astromech", "q7"},
+		{"astromech", "r2"},
+		{"astromech", "r4"},
+		{"battle", "b1"},
+		{"battle", "ig"},
+		{"probe", "viper"},
+		{"protocol", "000"},
+		{"protocol", "c3po"},
```

Comment:
I wonder if there's any remote chance of getting in trouble for naming C3PO in your code? You used nicknames for the astromechs, so maybe "3po" would be both safer and more consistent here in any case? 😆

Reply:
yeah I had a feeling the tm might be an issue; I'll change that